### PR TITLE
Feature/refactor mail function

### PR DESCRIPTION
**Pull request by robertbor**

### CODE REVIEW

File: omelet-core/src/main/java/omelet/mail/Email.java [hunk 1]
```diff
@@ -41,340 +34,371 @@
...
+        this.port = builder.port;
+        this.userName = builder.userName;
+        this.password = builder.password;
+        protocol = builder.protocol;
+        this.folderName = "Inbox";
+        connect(protocol);
+    }
+
+    private void connect(MailProtocol protocol) {
+        String hostPropertyKey;
```
Comment (kapilag) [inline omelet-core/src/main/java/omelet/mail/Email.java]: can we remove property from the variable name to something like hostKey ?

File: omelet-core/src/main/java/omelet/mail/Email.java [hunk 2]
```diff
@@ -41,340 +34,371 @@
...
+        String portPropertyKey;
+        String connectionTimeoutPropertyKey;
+        String timeoutPropertyKey;
+        Properties props = System.getProperties();
+        Session session;
+        switch (protocol) {
+            case POP3:
+                System.out.println("POP3");
```
Comment (kapilag) [inline omelet-core/src/main/java/omelet/mail/Email.java]: Please add Logger class , sonar will not be happy with this

File: omelet-core/src/main/java/omelet/mail/Email.java [hunk 3]
```diff
@@ -41,340 +34,371 @@
...
+                hostPropertyKey = "mail.pop3.host";
+                portPropertyKey = "mail.pop3.port";
+                props.setProperty("mail.store.protocol", "pop3s");
+                props.setProperty("mail.pop3.user", userName);
+                connectionTimeoutPropertyKey = "mail.pop3.connectiontimeout";
+                timeoutPropertyKey = "mail.pop3.timeout";
+                props.setProperty("mail.pop3.socketFactory.class", "javax.net.ssl.SSLSocketFactory");
+                props.setProperty("mail.pop3.socketFactory.fallback", "false");
+                break;
+            case IMAP:
```
Comment (kapilag) [inline omelet-core/src/main/java/omelet/mail/Email.java]: should we make small private methods for configuring property for smtp,pop3, imap to increase readability , just a suggestion though

File: omelet-core/src/main/java/omelet/mail/Email.java [hunk 4]
```diff
@@ -41,340 +34,371 @@
...
+    /***
+     * Return List of Message filter by Subject,From_ADD,To_ADDR
+     * Element 0 is the newest one!!
+     *
+     * @param emailFilter enum
+     * @param filterText
+     *            :text present in Subject of email
+     * @return list of messages
+     */
+    private List<Message> getMessages(EMAIL_FILTER emailFilter, String filterText) {
```
Comment (kapilag) [inline omelet-core/src/main/java/omelet/mail/Email.java]: Please change the method name to getEmails() to be consistent across other methods

File: omelet-core/src/main/java/omelet/mail/Email.java [hunk 5]
```diff
@@ -41,340 +34,371 @@
...
+            }
+        }
+        sw.stop();
+        LOGGER.info("Time Taken by Filter EmailBy Subjects is: "
+                + sw.elapsedTime(TimeUnit.SECONDS));
+        return returnMessage;
+    }
+
+    @Override
+    public boolean checkPatternInEmail(Message message, String patterToMatch) {
```
Comment (kapilag) [inline omelet-core/src/main/java/omelet/mail/Email.java]: method name to verifyPatternInEmail?

**Comment by robertbor** [comment]
Refactor complete....please have a look at it again.

**Pull request merged**